drivers: video: emul: simplify, avoid mission creep

josuah opened the pull request:
The general goal is making `video_emul_*` just enough complexity to test the APIs on CI.

- Instead of storing a full frame in the emulated imager, only store one line.
- Drop the test pattern generation, as `VIDEO_SW_GENERATOR` does it and no use in maintaining it twice.
- Other fixes and minor improvements.

[EDIT: updated the PR goals mid-way]

josuah committed:
drivers: video: emul: store only one line of data

Only store a single line of the full frame. This allows to support a
large enough frame size to remove the Kconfig option, which simplifies
the implementation (fewer checks needed).

Signed-off-by: Josuah Demangeon <me@josuah.net>

josuah committed:
drivers: video: emul_imager: remove dev->name from logs

Depending on the log mode selected, the device name will be visible
for every log message, so not useful to add it again in the log payload.
Various other log improvements also added.

Signed-off-by: Josuah Demangeon <me@josuah.net>

loicpoulain commented on tests/drivers/video/api/src/video_emul.c:
```diff
@@ -157,7 +157,7 @@ ZTEST(video_common, test_video_vbuf)
 
 	/* Allocate a buffer, assuming prj.conf gives enough memory for it */
-	vbuf = video_buffer_alloc(fmt.pitch * fmt.height, K_FOREVER);
+	vbuf = video_buffer_alloc(fmt.pitch * fmt.height, K_NO_WAIT);
```
This change is not explained in the commit message. Also I suppose the below assertion is then not valid anymore?

josuah replied:
Good point thank you, I will revisit it.

josuah replied:
This is now its [own commit](https://github.com/zephyrproject-rtos/zephyr/compare/da560abfeea958cd2cf7fe79dd5ececf5b0f7ed8..31d459b384335f40db64ae10b97409f299924529) proposing not to wait CI timeout when running out of video memory.

josuah committed:
drivers: video: emul_imager: deduplicate the registers

Deduplicate the registers of "modes": only a single definition of each:
pixelformats, resolutions, framerate.

Signed-off-by: Josuah Demangeon <me@josuah.net>

josuah committed:
drivers: video: emul_imager: remove the test pattern support

Remove the test pattern generator implemented inline in video_emul_imager,
as this feature is overlapping with VIDEO_SW_GENERATOR, and does not add
value to the driver.

Signed-off-by: Josuah Demangeon <me@josuah.net>

josuah committed:
drivers: video: api: do not wait the timeout

Fail immediately during tests rather than letting K_FOREVER trigger a CI
timeout.

Signed-off-by: Josuah Demangeon <me@josuah.net>

josuah committed:
drivers: video: emul_imager: use a "PRIV" video CID

Use a video control ID in the private range rather than using an existing
CID, as the emulated imager does not effectively applies these properties
to the image feed and are just for API test purpose.

Signed-off-by: Josuah Demangeon <me@josuah.net>
